pkgs/development/libraries/mesa/default.nix

```diff
@@ -6,9 +6,52 @@
 , libelf, libvdpau
 , libglvnd, libunwind
 , vulkan-loader, glslang
-, galliumDrivers ? ["auto"]
-# upstream Mesa defaults to only enabling swrast (aka lavapipe) on aarch64 for some reason, so force building the others
-, vulkanDrivers ? if (stdenv.isLinux && stdenv.isAarch64) then [ "swrast" "broadcom" "freedreno" "panfrost" ] else [ "auto" ]
+, galliumDrivers ?
+  if stdenv.isLinux then
+    [
+      "d3d12" # WSL emulated GPU (aka Dozen)
+      "iris" # new Intel
+      "kmsro" # helper driver for display-only devices
+      "nouveau" # Nvidia
+      "radeonsi" # new AMD (GCN+)
+      "r300" # very old AMD
+      "r600" # less old AMD
+      "swrast" # software renderer (aka LLVMPipe)
+      "svga" # VMWare virtualized GPU
+      "virgl" # QEMU virtualized GPU (aka VirGL)
+      "zink" # generic OpenGL over Vulkan, experimental
+    ]
+    ++ lib.optionals stdenv.isAarch64 [
+      "etnaviv" # Vivante GPU designs (mostly NXP/Marvell SoCs)
+      "freedreno" # Qualcomm Adreno (all Qualcomm SoCs)
+      "lima" # ARM Mali 4xx
+      "panfrost" # ARM Mali Midgard and up (T/G series)
+      "tegra" # Nvidia Tegra SoCs
+      "v3d" # Broadcom VC5 (Raspberry Pi 4)
+      "vc4" # Broadcom VC4 (Raspberry Pi 0-3)
+    ] ++ lib.optionals stdenv.isx86_64 [
+      "crocus" # Intel legacy, x86_64 only
+    ]
```

Comment:
If we're doing this, can we please take non-x86_64, non-aarch64 platforms into account by at least copying the defaults from mesa's meson.build?

Reply:
Why `else [ "auto" ]` doesn't cover those well?

Reply:
This is actually an exhaustive list. Literally everything that's currently available in Mesa gets built one way or the other. Probably won't stay that way, but it is at least for now.

Reply:
> Why `else [ "auto" ]` doesn't cover those well?

Not sure whether you mean "doesn't it currently …?", or "why doesn't it …?" but in case of the former, it doesn't, because the else only applies to non-Linux.